sled-agent/src/zone_bundle.rs

```diff
@@ -80,22 +80,37 @@ fn initialize_zfs_resources(log: &Logger) -> Result<(), BundleError> {
             // Additionally check for the zone-bundle-specific property.
             //
             // If we find a dataset that matches our names, but which _does not_
-            // have such a property, we'll panic rather than possibly deleting
-            // user data.
+            // have such a property (or has in invalid property), we'll log it
+            // but avoid deleting the snapshot.
             let name = snap.to_string();
-            let value =
-                Zfs::get_oxide_value(&name, ZONE_BUNDLE_ZFS_PROPERTY_NAME)
-                    .unwrap_or_else(|_| {
-                        panic!(
-                            "Found a ZFS snapshot with a name reserved for \
-                            zone bundling, but which does not have the \
-                            zone-bundle-specific property. Bailing out, \
-                            rather than risking deletion of user data. \
-                            snap_name = {}, property = {}",
-                            &name, ZONE_BUNDLE_ZFS_PROPERTY_NAME
-                        )
-                    });
-            assert_eq!(value, ZONE_BUNDLE_ZFS_PROPERTY_VALUE);
+            let Ok([value]) = Zfs::get_values(
+                &name,
+                &[ZONE_BUNDLE_ZFS_PROPERTY_NAME],
+                Some(illumos_utils::zfs::PropertySource::Local),
+            ) else {
+                warn!(
+                    log,
+                    "Found a ZFS snapshot with a name reserved for zone \
+                    bundling, but which does not have the zone-bundle-specific \
+                    property. Bailing out, rather than risking deletion of \
+                    user data.";
+                    "snap_name" => &name,
+                    "property" => ZONE_BUNDLE_ZFS_PROPERTY_NAME
```

Comment:
I realize this already landed, but this log should probably include the error from `get_values`, right? It might be that we don't have the property (as the current log message says), but it could also be something like "`zfs get` failed outright".